File: LABA5/parserPack/GenreParser.java
package parserPack;

import elementsPack.MovieGenre;
import exceptionsPack.NullFieldException;

public class GenreParser {
    public static MovieGenre parse(String line) throws NullFieldException {
        line = line.toLowerCase();
        if (line.isEmpty()) throw new NullFieldException();
        return line.equals("drama") ? MovieGenre.DRAMA
                : line.equals("fantasy") ? MovieGenre.FANTASY
                : line.equals("tragedy") ? MovieGenre.TRAGEDY
                : line.equals("western") ? MovieGenre.WESTERN
                : null;
    }
}

File: LABA5kt/src/parserPack/GenreParser.kt
package parserPack

import elementsPack.MovieGenre
import exceptionsPack.NullFieldException
import java.util.*

object GenreParser {
    /**
     * Parses the given string to a MovieGenre.
     *
     * @param  line  the string to be parsed
     * @return      the parsed MovieGenre
     */
    @Throws(NullFieldException::class)
    fun parse(line: String): MovieGenre {
        if (line.isEmpty()) throw NullFieldException()
        return when (line.lowercase(Locale.getDefault())) {
            "drama" -> MovieGenre.DRAMA
            "fantasy" -> MovieGenre.FANTASY
            "tragedy" -> MovieGenre.TRAGEDY
            "western" -> MovieGenre.WESTERN
            else -> null!!
        }
    }
}